### ISSUE TITLE
Table.to_pandas() failed with some integer masked column, eg. those of '>i4' type

### ISSUE THREAD

**Issue by orionlee** [astropy/astropy#11645]
### Description

Calling `Table.to_pandas()` yields `TypeError: data type '>I4' not understood` in a specific situation:
1. The table has `MaskedColumn`
2. The masked column is of certain integer type, e.g., `>i4` or `u4`  (`int32` does not cause error)
3. some value is masked

In such cases, calling `Table.to_pandas()` yields `TypeError`.

See https://github.com/lightkurve/lightkurve/issues/1043 for a downstream issue that is affected by this one.

Workaround: before calling `to_pandas()`, change the dtype to comptaible types that work, e.g., for `>i4`, change the dtype to `np.int32`

### Steps to Reproduce

The following snippets would yield `TypeError`
```python
from astropy.table import Table

tab = Table(masked=True)
tab['CADENCENO'] = [0, 1, 2, 3]
tab['CADENCENO'].dtype = np.dtype('>i4') # emulate the dtype from reading TESS fits file
tab['CADENCENO'].mask[2] = True
print(tab['CADENCENO'].dtype)
tab.to_pandas()

```

### System Details
Windows-10-10.0.19041-SP0
Python 3.7.10 (default, Feb 26 2021, 13:06:18) [MSC v.1916 64 bit (AMD64)]
Numpy 1.19.5
astropy 4.1
Scipy 1.5.4
Matplotlib 3.3.4
pandas 1.1.5

**Comment by orionlee** [comment]
The root cause:

In `to_pandas()` code:
https://github.com/astropy/astropy/blob/0b75d7fe054677840b1890c795a81f69bd3067a0/astropy/table/table.py#L3428-L3431

If the `dtype` of a `MaskedColumn` is '>i4', 
- the codes will try to create a panda Series with dtype `>I4` (capital I). 
- Pandas reject the type, ultimately with the `TypeError`

**Comment by pllim** [comment]
Looks like @bmorris3 added that logic in #9541 to solve #9280. cc @taldcroft

**Comment by taldcroft** [comment]
When I run the steps to reproduce it gives an exception before getting to the pandas bit:
```
tab = Table(masked=True)
tab['CADENCENO'] = [0, 1, 2, 3]
----> tab['CADENCENO'].dtype = np.dtype('>i4') # emulate the dtype from reading TESS fits file

ValueError: cannot reshape array of size 4 into shape (8,)
```
If instead I do the following it works as expected:
```
In [7]: from astropy.table import Table, MaskedColumn
   ...: 
   ...: tab = Table(masked=True)
   ...: tab['CADENCENO'] = np.array([0, 1, 2, 3], dtype='>i4')
   ...: tab['CADENCENO'].mask[2] = True
   ...: print(tab['CADENCENO'].dtype)
   ...: tab.to_pandas()
   ...: 
>i4
Out[7]: 
   CADENCENO
0          0
1          1
2       <NA>
3          3
```

**Comment by orionlee** [comment]
For the test codes (both my version and the @taldcroft 's variant), both failed `TypeError: data type '>I4' not understood` in my environment.

1. I'll try to run it against the astropy's `main` and report back.
2. @taldcroft Could you also tell me the environment you use (OS, astropy version, pandas version, etc.)>

**Comment by taldcroft** [comment]
@orionlee - apologies. I was running in current `main` on Mac.  I can reproduce getting that `TypeError` you reported using astropy 4.2. So from what I can see this has been fixed on `main` which will be released relatively soon in 4.3.

**Comment by orionlee** [comment]
@taldcroft You're right - it's fixed in PR #11294. Closing it.
